openstates/ar/bills.py

```diff
@@ -57,29 +61,31 @@ def scrape_bill(self, chamber, session):
                 'R': 'resolution',
                 'JR': 'joint resolution',
                 'CR': 'concurrent resolution',
-                'MR': 'memorial resolution',
-                'CMR': 'concurrent memorial resolution'}[type_spec]
+                'MR': 'memorial',
+                'CMR': 'concurrent memorial'}[type_spec]
 
             if row[-1] != self.slug:
                 continue
 
-            bill = Bill(session, chamber, bill_id, row[3], type=bill_type)
+            bill = Bill(bill_id, legislative_session=session,
+                        chamber=chamber, title=row[3], classification=bill_type)
             bill.add_source(url)
 
             primary = row[11]
             if not primary:
                 primary = row[12]
-            if primary:
-                bill.add_sponsor('primary', primary)
 
+            if primary:
+                bill.add_sponsorship(primary, classification='primary',
+                                     entity_type='person', primary=True)
             # ftp://www.arkleg.state.ar.us/Bills/
             # TODO: Keep on eye on this post 2017 to see if they apply R going forward.
             session_code = '2017R' if session == '2017' else session
 
             version_url = ("ftp://www.arkleg.state.ar.us/Bills/"
                            "%s/Public/%s.pdf" % (
                                session_code, bill_id.replace(' ', '')))
-            bill.add_version(bill_id, version_url, mimetype='application/pdf')
+            bill.add_version_link(bill_id, version_url, media_type='application/pdf')
 
             self.scrape_bill_page(bill)
```

Comment:
It looks like `scrape_bill_page` is now a generator, so we need to `yield from` this call to save votes.